### PR TITLE
 fix: handle mantissa round overflow

### PR DESCRIPTION
**Pull request by xin3he**
## Description

This pull request improves the correctness and robustness of the E5M3 floating-point quantization implementation and its CUDA extension, focusing on proper handling of mantissa rounding and exponent carry, as well as adding comprehensive tests for these behaviors. The main changes include a refactor of the `float_to_e5m3_frexp` function to handle rounding edge cases, updates to the CUDA kernel for consistency, and new unit tests to verify correct behavior at rounding boundaries and with special values.

**E5M3 quantization logic improvements:**

* Refactored `float_to_e5m3_frexp` in `nvfp.py` to correctly handle mantissa rounding overflow by carrying to the exponent, saturating outputs, and handling NaN/Inf values according to the E5M3 spec.
* Updated CUDA kernel `qdq_kernel` in `cute_nvfp4_e5m3.py` to match the new rounding and carry logic, ensuring consistency between CPU and GPU implementations.

**Testing enhancements:**

* Added unit tests in `test_nvfp.py` to verify mantissa rounding overflow, rounding boundaries, saturation, and correct handling of NaN/Inf for `float_to_e5m3_frexp`.
* Updated and added CUDA-specific tests in `test_nvfp4_e5m3_cute.py` to ensure the CUDA path matches the reference implementation, especially at rounding boundaries and when mantissa overflow occurs.

## Type of Change

Bug fix

## Related Issues

Fixes or relates to #

## Checklist Before Submitting

- [ ] My code has been tested locally.
- [ ] Documentation has been updated as needed.
- [ ] New or updated tests are included where applicable.
- [ ] The CUDA CI has passed. You can trigger it by commenting `/azp run Unit-Test-CUDA-AutoRound`.

### CODE REVIEW

**Comment by xin3he** [comment]
- 数值例子（直观）：
      1. scale = 1.9375，frexp → mantissa=0.96875，exponent=1：
      2. (mantissa-0.5)*16 = 7.5 → round → 8
      3. 正确 RNE（carry）：m→0，exponent→2 → 解码得 2.0
      4. clamp=7（错误）：m→7，exponent仍1 → 解码得 1.875
      5. 差值：2.0 vs 1.875，约 6.67% 相对误差
cc @wenhuach21

**Comment by xin3he** [comment]
Comparing with https://github.com/intel/auto-round/pull/2123, accuracy increased from 0.7196 to 0.7217

File: auto_round/data_type/nvfp.py [hunk 1]
```diff
@@ -125,34 +125,20 @@ def nv_fp4_with_static_gs(tensor, bits=4, group_size=16, v=0, tensor_max=None, *
...
-    out_vals[subnormal_mask] = e5m3_sub
-
-    e5m3[mask] = out_vals
-    return e5m3
+    input_fp32 = x.to(torch.float32)
+    finite = torch.nan_to_num(input_fp32, nan=0.0, posinf=FLOAT8_UE5M3_MAX, neginf=0.0).clamp_(0.0, FLOAT8_UE5M3_MAX)
+
+    mantissa, exponent = torch.frexp(finite.clamp_min(2**-14))
+    m3 = torch.round((mantissa - 0.5) * 16).to(torch.int32)
+    carry = m3 == 8
```
Comment (wenhuach21) [inline auto_round/data_type/nvfp.py:133]: Very nice catch! Thanks

**Review by wenhuach21 — APPROVED** [review]
